test/integration/ads_integration_test.cc

```diff
@@ -992,4 +992,379 @@ TEST_P(AdsIntegrationTestWithRtdsAndSecondaryClusters, Basic) {
   testBasicFlow();
 }
 
+static std::string adsV3IntegrationConfig(const std::string& api_type) {
+  // Note: do not use CONSTRUCT_ON_FIRST_USE here!
+  return fmt::format(R"EOF(
+dynamic_resources:
+  lds_config:
+    resource_api_version: V3
+    ads: {{}}
+  cds_config:
+    resource_api_version: V3
+    ads: {{}}
+  ads_config:
+    transport_api_version: V3
+    api_type: {}
+    set_node_on_first_message_only: false
+static_resources:
+  clusters:
+    name: dummy_cluster
+    connect_timeout:
+      seconds: 5
+    type: STATIC
+    load_assignment:
+      cluster_name: dummy_cluster
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: 127.0.0.1
+                port_value: 0
+    lb_policy: ROUND_ROBIN
+    http2_protocol_options: {{}}
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: 0
+)EOF",
+                     api_type);
+}
+
+// Check if EDS cluster defined in file is loaded before ADS request and used as xDS server
+class AdsClusterV3 : public Grpc::DeltaSotwIntegrationParamTest, public HttpIntegrationTest {
```

Comment:
What this test verifies looks totally valid, is it possible to do this without as much copy+paste? It will be hard to review and maintain this long term. Since all that differs is the resource, it should just be possible to parameterize the cluster and CLA generation methods in ADS integration test with version.

Reply:
will address this

Reply:
@htuch i made AdsIntegrationTest parameterized by ApiVersion which avoids parametrizing each method, let me know if this is ok.

Do you think `sendDiscoveryResponse` methods should be parametrized as well?